includes/Modules/Analytics.php

```diff
@@ -145,6 +146,26 @@ function( $old_value, $new_value ) {
 			2
 		);
 
+		// Ensure both Analytics modules always reference the same owner.
+		add_filter(
+			'pre_update_option_' . Analytics_4_Settings::OPTION,
+			function( $new_value, $old_value ) {
+				if ( $old_value['ownerID'] !== $new_value['ownerID'] ) {
+					$settings = $this->get_settings()->get();
+
+					if ( $settings['ownerID'] && $new_value['ownerID'] !== $settings['ownerID'] ) {
+						$this->get_settings()->merge(
+							array( 'ownerID' => $new_value['ownerID'] )
+						);
+					}
+				}
+
+				return $new_value;
+			},
+			20,
+			2
+		);
```

Comment:
I know this is the approach is what was suggested in the IB, but I find it _slightly_ confusing that the update filters for one module are in the other.

I'd sort of expect that the code that handles GA4 updates to be in the GA4 module, and then apply to other modules.

Is there any issue with that approach? It just threw me off seeing this in the `Analytics` code and the opposite in the `Analytics4` code 🤔